package/gluon-scheduled-domain-switch/luasrc/usr/bin/gluon-switch-domain

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/lua
+
+local uci = require('simple-uci').cursor()
+local unistd = require 'posix.unistd'
+local site = require 'gluon.site'
+
+-- Returns true if given domain is present on device
+function domain_exists(domain_name)
+	return unistd.access("/lib/gluon/domains/"..domain_name..".json")
+end
+
+-- Returns true if node was offline long enough to perform domain switch
+function switch_after_min_reached()
+	if not unistd.access("/tmp/gluon_gw_unreach") then
+		return false
+	end
+
+	local current_epoch = os.time()
+	local switch_after_sec = site.domain_switch.switch_after_offline_mins() * 60
+
+	local current_uptime_handle = io.popen('cat /proc/uptime | cut -d "." -f1')
+	local current_uptime = tonumber(current_uptime_handle:read('*all'))
+	current_uptime_handle:close()
+
+	local f = io.open("/tmp/gluon_gw_unreach")
+	if not f then
+		return false
+	end
+	local offline_since = tonumber(f:read('*a'))
+	f:close()
+
+	local offline_time_sec = current_uptime - offline_since
+
+	if offline_time_sec > switch_after_sec then
+		return true
+	end
+	return false
+end
+
+-- Returns true in case switch time has passed
+function switch_time_passed()
+	local current_epoch = os.time()
+	local switch_time = site.domain_switch.switch_time()
+
+	if switch_time < current_epoch then
+		return true
+	end
+
+	return false
+end
+
+if site.domain_switch() == nil or site.domain_switch.target_domain() == nil or site.domain_switch.switch_after_offline_mins() == nil or site.domain_switch.switch_time() == nil then
+	-- Switch not applicable for current domain
+	print("No domain switch defined for the current domain.")
+	return 1
+end
+
+local current_domain = uci:get("gluon", "core", "domain")
+local target_domain = site.domain_switch.target_domain()
+
+if not domain_exists(target_domain) then
+	-- Domain does not exist
+	print("Domain '" .. target_domain .. "' does not exist.")
+	return 1
+end
+
+if target_domain == current_domain then
+	-- Current and target domain are equal
+	print("Domain '" .. target_domain .. "' equals current domain.")
+	return 0
+end
+
+if not switch_after_min_reached() and not switch_time_passed() then
+	-- Neither switch-time passed nor switch_after_min reached
+	return 0
+end
+
+uci:set("gluon", "core", "domain", target_domain)
+uci:commit("gluon")
+
+local gluon_reconfigure_handle = io.popen("gluon-reconfigure")
+gluon_reconfigure_handle:read('*all')
+gluon_reconfigure_handle:close()
```

Comment:
Use os.execute for gluon-reconfigure as well.

Reply:
This does not work as os.execute does not wait for termination of gluob-reconfigure.

Reply:
It does - os.execute returns the exit code of the passed command.